xapian-core/include/xapian/cluster.h

```diff
@@ -28,14 +28,38 @@
 #endif
 
 #include <xapian/mset.h>
+#include <xapian/queryparser.h>
 #include <xapian/types.h>
 #include <xapian/visibility.h>
 
 #include <unordered_map>
+#include <unordered_set>
 #include <vector>
 
 namespace Xapian {
 
+/// Stopper subclass which checks for both stemmed and unstemmed stopwords
+class XAPIAN_VISIBILITY_DEFAULT StemStopper : public Xapian::Stopper {
+    std::unordered_set<std::string> stop_words;
+    Xapian::Stem stemmer;
+
+  public:
+    /** Constructor
+     *
+     *  @param stemmer	The Xapian::Stem object to set.
+     */
+    explicit StemStopper(const Xapian::Stem &stemmer);
+
+    std::string get_description() const;
+
+    bool operator()(const std::string & term) const {
+	return stop_words.find(term) != stop_words.end();
+    }
+
+    /// Add a single stop word and its stemmed equivalent
+    void add(const std::string &term);
+};
+
```

Comment:
Yes as @ojwb mentioned, StemStopper is used to allow filtering of stopwords which are stemmed when we are creating the document term lists. For creating document term lists, we are using two types of dimensionality reduction - stopword removal and stemming. Since document termlists already contain stemmed terms, that isn't a problem but identifying stopwords which are stemmed is what is needed. Thus if a user inputs the stopwords that they want to filter with, we can stem and remove the stopwords, thus giving us a smaller and more representative termlist.